libs/partners/google-vertexai/langchain_google_vertexai/vectorstores.py

```diff
@@ -0,0 +1,613 @@
+    using this module.
+
+    See usage in docs/integrations/vectorstores/google_vertex_ai_vector_search.ipynb
+
+    Note that this implementation is mostly meant for reading if you are
+    planning to do a real time implementation. While reading is a real time
+    operation, updating the index takes close to one hour."""
+
+    def __init__(
+        self,
+        project_id: str,
+        index: MatchingEngineIndex,
+        endpoint: MatchingEngineIndexEndpoint,
+        embedding: Embeddings,
+        gcs_client: storage.Client,
+        gcs_bucket_name: str,
+        credentials: Optional[Credentials] = None,
+        *,
+        document_id_key: Optional[str] = None,
+    ):
+        """Google Vertex AI Vector Search (previously Matching Engine)
+         implementation of the vector store.
+
+        While the embeddings are stored in the Matching Engine, the embedded
+        documents will be stored in GCS.
+
+        An existing Index and corresponding Endpoint are preconditions for
+        using this module.
+
+        See usage in
+        docs/integrations/vectorstores/google_vertex_ai_vector_search.ipynb.
+
+        Note that this implementation is mostly meant for reading if you are
+        planning to do a real time implementation. While reading is a real time
+        operation, updating the index takes close to one hour.
+
+        Attributes:
+            project_id: The GCS project id.
+            index: The created index class. See
+                ~:func:`VertexAIVectorSearch.from_components`.
+            endpoint: The created endpoint class. See
+                ~:func:`VertexAIVectorSearch.from_components`.
+            embedding: A :class:`Embeddings` that will be used for
+                embedding the text sent. If none is sent, then the
+                multilingual Tensorflow Universal Sentence Encoder will be used.
+            gcs_client: The GCS client.
+            gcs_bucket_name: The GCS bucket name.
+            credentials (Optional): Created GCP credentials.
+            document_id_key (Optional): Key for storing document ID in document
+                metadata. If None, document ID will not be returned in document
+                metadata.
+        """
+        super().__init__()
+        self._validate_google_libraries_installation()
+
+        self.project_id = project_id
+        self.index = index
+        self.endpoint = endpoint
+        self.embedding = embedding
+        self.gcs_client = gcs_client
+        self.credentials = credentials
+        self.gcs_bucket_name = gcs_bucket_name
+        self.document_id_key = document_id_key
+
+    @property
+    def embeddings(self) -> Embeddings:
+        return self.embedding
+
+    def _validate_google_libraries_installation(self) -> None:
+        """Validates that Google libraries that are needed are installed."""
+        try:
+            from google.cloud import aiplatform, storage  # noqa: F401
```

Comment:
we can move it on top of the file now, `aiplatform` is a requirement for the package